### PR TITLE
fix(gainsight): replace myAxios utility with handleHttpRequest utility

### PR DESCRIPTION
**Pull request by mihir-4116**
## What are the changes introduced in this PR?
- This pr replaces myAxios utility with handleHttpRequest utility for gainsight destination

## What is the related Linear task?

Resolves INT-1906

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3241/test-report.html

File: src/v0/destinations/gainsight/util.js [hunk 1]
```diff
@@ -1,134 +1,132 @@
...
-    }
-    throw new NetworkError(`failed to search group ${errMessage}`, errorStatus, {
-      [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(errorStatus),
-    });
   }
 
-  if (!resp || !resp.data || resp.status !== 200) {
-    throw new RetryableError('failed to search group');
+  if (!processedResponse || !processedResponse.response || processedResponse.status !== 200) {
+    throw new RetryRequestError('failed to search group');
```
Comment (sanpj2292) [inline src/v0/destinations/gainsight/util.js:43]: `RetryRequestError` sets the status-code to be `809` and not `500`. We should use `RetryableError` in this case

File: src/v0/destinations/gainsight/util.js [hunk 2]
```diff
@@ -1,134 +1,132 @@
...
-    }
-    throw new NetworkError(`failed to create group ${errMessage}`, errorStatus, {
-      [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(errorStatus),
-    });
   }
 
-  if (!resp || !resp.data || resp.status !== 200) {
-    throw new RetryableError('failed to create group');
+  if (!processedResponse || !processedResponse.response || processedResponse.status !== 200) {
+    throw new RetryRequestError('failed to create group');
```
Comment (sanpj2292) [inline src/v0/destinations/gainsight/util.js:83]: https://github.com/rudderlabs/rudder-transformer/pull/3241/files#r1562131204

File: src/v0/destinations/gainsight/util.js [hunk 3]
```diff
@@ -1,134 +1,132 @@
...
-    }
-    throw new NetworkError(`failed to update group ${errMessage}`, errorStatus, {
-      [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(errorStatus),
-    });
   }
 
-  if (!resp || !resp.data || resp.status !== 200) {
-    throw new RetryableError('failed to update group');
+  if (!processedResponse || !processedResponse.response || processedResponse.status !== 200) {
+    throw new RetryRequestError('failed to update group');
```
Comment (sanpj2292) [inline src/v0/destinations/gainsight/util.js:126]: https://github.com/rudderlabs/rudder-transformer/pull/3241/files#r1562131204

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Review by sanpj2292 — DISMISSED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**